### PR TITLE
ATMOS SPEED ZOOM ZOOM WEEE Optimises two loops that Dunc missed.

### PR DESCRIPTION
**Pull request by RemieRichards**
@duncathan nice try :3

### CODE REVIEW

**Comment by Iamgoofball** [comment]
is this seriously faster?

**Comment by MrStonedOne** [comment]
That is a poor name


How are the people on the server suppose to know that atmos code is getting faster?

fucking close, rename, and re-open this. RIGHT NOW!

=P

**Comment by RemieRichards** [comment]
@Iamgoofball yes

**Comment by Shapsy** [comment]
Why do these PR's always have such technical names, can someone explain what "ZOOM ZOOM WEEE" means in layman's terms?

**Comment by Cheridan** [comment]
@Snapsy Please refer to this tutorial video: https://youtu.be/Eo_gJwXxshQ

**Comment by FlavoredCactus** [comment]
Question, How is this faster? I'm no wizard but i think looking through EVERY VAR in a foreach loop would take slightly longer then a specific type. How much of a difference is this anyway?

**Comment by xxalpha** [comment]
Byond will do istype() checks when you specify a type in the for header, it will not do so if you don't specify any type. If you know that there is only objects of one type in a list, it's faster to type cast the objects than it is to specify a type in the for header.

**Comment by FlavoredCactus** [comment]
interesting

**Comment by RemieRichards** [comment]
Yeah, @xxalpha explained it perfectly.

**Comment by duncathan** [comment]
Dammit :v
:+1:
